spring-integration-core/src/main/java/org/springframework/integration/mapping/support/JsonHeaders.java

```diff
@@ -105,9 +106,9 @@ else if (classValue != null) {
 	 * @return the {@link ResolvableType} based on provided class components
 	 * @since 5.2.4
 	 */
-	public static ResolvableType buildResolvableType(@Nullable Class<?> targetClass, @Nullable Class<?> contentClass,
+	@Contract("_, !null, _ -> !null; _, _, !null -> !null")
```

Comment:
This is wrong.
The method does not return `null` at all.
See `getClassForValue()`.
That's where the `@Contract` has to be.